source/sdk/structs/Stack.ooc

```diff
@@ -1,73 +1,54 @@
-import structs/ArrayList
-
-Stack: class <T> extends BackIterable<T> {
-    data: ArrayList<T>
-    
-    size: SSizeT {
-        get {
-            data != null ? data size : 0		
-        }
-    }
-    
-    init: func {
-        data = ArrayList<T> new()
-    }
-
-    free: override func {
-      this data free()
-      super()
-    }
-
-    push: func(element: T) {
-        data add(element)
-    }
-    
-    pop: func -> T {
-        if (empty?())
-            Exception new(This, "Trying to pop an empty stack.") throw()
-            
-        return data removeAt(lastIndex())
-    }
-    
-    peek: func -> T {
-        if (empty?())
-            Exception new(This, "Trying to peek an empty stack.") throw()
-                
-        return data last()
-    }
-
-    peek: func ~index (index: SSizeT) -> T {
-        mysize := data size
-        if (index < 1)
-            Exception new(This, "Trying to peek(%d)! index must be >= 1 <= size" format(index)) throw()
-
-        if (index > mysize)
-            Exception new(This, "Trying to peek(%d) a stack of size %d" format(index, mysize)) throw()
-
-        return data get(mysize - index)
-    }
-
-    indexOf: func(element: T) -> SSizeT {
-        return data indexOf(element)
-    }
-	
-    getSize: func -> SSizeT {	
-        return data size
-    }
-    
-    empty?: func -> Bool {
-        return data empty?()
-    }
-    
-    lastIndex: func -> Int {
-        return size - 1
-    }
-
-    clear: func {
-        data clear()
-    }
-
-    iterator: func -> BackIterator<T> { data iterator() }
-
-    backIterator: func -> BackIterator<T> { data backIterator() }
+Stack: class <T> {
+	_data: T*
+	_size: SizeT = 0
+	_capacity: SizeT = 8
+	size: SizeT {
+		get {
+			this _size
+		}
+	}
```

Comment:
> Alias of one of the fundamental unsigned integer types.
>size_t is the type returned by the sizeof operator

Also it is heavily use in C++ STL. So I guess it is platform dependent the same way as `sizeof` is